src/aap_eda/api/views/project.py

```diff
@@ -196,14 +197,18 @@ def partial_update(self, request, pk):
     @action(
         methods=["post"],
         detail=True,
-        rbac_action=Action.UPDATE,
+        rbac_action=Action.SYNC,
     )
     @transaction.atomic
     def sync(self, request, pk):
-        try:
-            project = self.get_queryset().select_for_update().get(pk=pk)
-        except models.Project.DoesNotExist:
-            raise api_exc.NotFound
+        # we have to use filter_queryset here instead of get_object in order
+        # to use select_for_update to lock project instance to update
+        # TODO: check how important is using select_for_update
+        queryset = self.filter_queryset(
+            self.get_queryset().select_for_update()
+        )
+        project = get_object_or_404(queryset, id=pk)
+        self.check_object_permissions(request, project)
```

Comment:
@Alex-Izquierdo @AlanCoding I'm not sure how important using `select_for_update()` method here is, if it could be removed then these 3 lines could be replaced with `get_object()`.

Reply:
It is almost certainly due to the `Conflict` exception below. The usage looks generally correct, since this is a function in a transaction. It loads `import_state` after entering, checks the data for some condition, and then sets `import_state` after starting. This is how you would be 100% sure 2 things didn't run at the same time.

You generally don't need to lump permission checks together with that, and `select_for_update` should be as limited in scope as humanly possible. I might suggest moving `transaction.atomic` from a decorator to a context manager after the permission checks. Then do `project.refresh_from_db(fields=['import_state'])` and the conflict check, and the save.

Reply:
I would say the lock is necessary to be sure there are no race conditions with rq jobs.

Reply:
@Alex-Izquierdo Pushed new changes that does correct permission checking without breaking the existing implementation as much as I can. @AlanCoding looks like we still want to keep `select_for_update` in order to lock the rows, and thus we cannot move the transaction to a context manager, because `select_for_update` has to be inside a transaction. Please let me know if I missed or misunderstood something.